lexicon/providers/euserv.py

```diff
@@ -0,0 +1,262 @@
+        LOGGER.debug('create_record response: %s', response)
+
+        return True
+
+    def _list_records(self, rtype=None, name=None, content=None):
+        LOGGER.info('Listing records for type=%s, name=%s, content=%s', rtype, name, content)
+
+        query_params = {'dns_records_load_only_for_dom_id': self.domain_id}
+
+        if rtype:
+            query_params['dns_records_load_type'] = rtype
+
+        if name:
+            query_params['dns_records_load_subdomain'] = self._subdomain_name(name)
+
+        if content:
+            query_params['dns_records_load_content'] = content
+
+        payload = self._get('kc2_domain_dns_get_records', query_params)
+
+        response = payload['result']['domains'][0]
+
+        LOGGER.debug('list_records raw response: %s', response)
+
+        records = []
+
+        if 'dns_records' in response:
+            for record in response['dns_records']:
+                processed_record = {
+                    'type': record['type']['value'],
+                    'name': record['name']['value'],
+                    'ttl': record['ttl']['value'],
+                    'content': record['content']['value'],
+                    'id': record['id']['value'],
+                    'priority': record['prio']['value']
+                }
+
+                records.append(processed_record)
+
+        return records
+
+    def _update_record(self, identifier, rtype=None, name=None, content=None):
+        if identifier is None:
+            LOGGER.info('No identifier for record provided, trying to find it...')
+            identifier = self._find_record_identifier(rtype, name, None)
+
+        data = {'dns_record_id': identifier}
+
+        if rtype:
+            data['type'] = rtype
+
+        if name:
+            data['subdomain'] = self._subdomain_name(name)
+
+        if content:
+            data['content'] = content
+
+        self._add_ttl(data)
+        self._add_priority(data)
+
+        response = self._get('kc2_domain_dns_set', data)
+        LOGGER.debug('update_record: %s', response)
+
+        return True
+
+    def _delete_record(self, identifier=None, rtype=None, name=None, content=None):
+        record_ids = []
+
+        if identifier is None:
+            LOGGER.info('No identifier for record provided, trying to find it...')
+            records = self._list_records(rtype, name, content)
+            record_ids = [record['id'] for record in records]
+        else:
+            record_ids.append(identifier)
+
+        LOGGER.debug('record IDs to be deleted: %s', record_ids)
+
+        for record_id in record_ids:
+            data = {'dns_record_id': record_id}
+
+            response = response = self._get('kc2_domain_dns_remove', data)
+            LOGGER.debug('delete_record: %s', response)
+
+        return True
+
+    # Helpers
+    # url param used as subaction
+    def _request(self, action='GET', url='/', data=None, query_params=None):
```

Comment:
I would prefer we do not mutate the contract around `_request`. Indeed this function is part of the `BaseProvider`, and is overridden by all providers, giving a global consistency.

Technically, you are always using `url=/`, and so you never need to modify it. Could you pass the subaction directly in the `query_params` instead?

Reply:
Good point, that makes sense. I changed it.